### PR TITLE
parser: support named struct types as auth handler params

### PR DESCRIPTION
**Pull request by eandre**

### CODE REVIEW

File: parser/svc.go [hunk 1]
```diff
@@ -442,15 +444,20 @@ func (p *parser) validateAuthHandler(h *est.AuthHandler) {
...
-	// Second param must be string
-	tok, _ := getField(params, 1)
-	if id, ok := tok.Type.(*ast.Ident); !ok || id.Name != "string" {
-		p.err(tok.Type.Pos(), "second parameter must be of type string"+sigHint)
-		return
-	} else if pkgNames.Decls["string"] != nil {
-		p.err(tok.Type.Pos(), "second parameter must be of type string (local name shadows builtin)"+sigHint)
-		return
+	// Second param must be string or named type pointing to a struct
+	authInfo, _ := getField(params, 1)
```
Comment (ekerfelt) [inline parser/svc.go:448]: Maybe we should require a pointer to a struct?

Comment (eandre) [reply parser/svc.go:448]: Haha yes, except we don't yet support pointers :)

**Review by DomBlack — APPROVED** [review]

File: parser/svc.go (hunk 1 above)
Comment (DomBlack) [reply parser/svc.go:448]: 🧂

Comment (ekerfelt) [reply parser/svc.go:448]: But this is the ast! You can just check that the authInfo.Type is an ast.StarExpr (if not string)

Comment (eandre) [reply parser/svc.go:448]: Oh true :) I wasn't sure if you were trolling!

Comment (ekerfelt) [reply parser/svc.go:448]: Me, troll, on a Tuesday?

**Pull request merged**